src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,87 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
 
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String SUPPLY = "supply";
+    private int buyAmount;
+    private String[] statistics;
+    private int supplyAmount;
+    private int totalAmount;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readStatistic(fromFileName);
+        addStatistic();
+
+        File report = new File(toFileName);
+        BufferedWriter bufferedWriter = null;
+
+        try {
+            bufferedWriter = new BufferedWriter(new FileWriter(report, true));
+            for (String stat : statistics) {
+                bufferedWriter.write(stat);
+                bufferedWriter.flush();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot write data to file", e);
+        } finally {
+            try {
+                bufferedWriter.close();
+            } catch (IOException e) {
+                throw new RuntimeException("Cannot write data to file", e);
+            }
+        }
+    }
+
+    public void readStatistic(String fromFileName) {
+        File file = new File(fromFileName);
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String inputData = bufferedReader.readLine();
```

Comment:
Optional, but we can avoid creating redundant variable 
```suggestion
        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(new File(fromFileName)))) {
            String inputData = bufferedReader.readLine();
```

Reply:
I have added an suggested piece of code